### PR TITLE
Raise exception when src_indices shape does not match target input shape

### LINKED ISSUE #1479: Exception should be raised for connection where src_indices shape does not match target input shape.
### Summary of Issue

Exception should be raised for connection where src_indices shape does not match target input shape.

### Issue Type

- [ x] Bug
- [ ] Enhancement  
      POEM #___ <N/A or POEM number at https://github.com/OpenMDAO/POEMs>

### Description

See example.  Must be a hole in our check logic.

### Example

https://gist.github.com/Kenneth-T-Moore/08674a0c0f41dd380978ba72f2f72efb

### Environment

<For bugs, otherwise N/A>

Operating System: <i.e. OS X 10.14.6, Windows 10, Ubuntu 16.04>

Python environment: <i.e. Anaconda Python 3.7.1>

Packages: <versions of Dymos, OpenMDAO, and other pertinent Python packages. You can just paste the output of `pip freeze` or `conda list` here.>

### PR DESCRIPTION
**Pull request by DKilkenny**
### Summary

Raising exception on connections where the src_indices don't match the target input shape

### Related Issues

- Resolves #1479

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: openmdao/core/group.py [hunk 1]
```diff
@@ -1302,6 +1302,15 @@ def _setup_connections(self):
                                         else:
                                             simple_warning(msg)
 
+                            for i in src_indices:
```
Comment (Kenneth-T-Moore) [inline openmdao/core/group.py:1305]: Can you move your implementation over to the `auto_ivc` branch, and issue the pull request there?

Comment (DKilkenny) [reply openmdao/core/group.py:1423]: Moved

File: openmdao/core/group.py [hunk 2]
```diff
@@ -1420,6 +1420,15 @@ def _setup_connections(self):
...
+                                if len(i.shape) != len(in_shape) and src_indices.shape != in_shape:
+                                    msg = f"{self.msginfo}: src_indices shape {src_indices.shape} does not match {abs_in} shape " + \
```
Comment (Kenneth-T-Moore) [inline openmdao/core/group.py:1425]: I don't understand the loop over src_indices.  Can't you just check if src_indices.shape != in_shape?

Comment (DKilkenny) [reply openmdao/core/group.py]: Removed

File: openmdao/core/tests/test_group.py [hunk 3]
```diff
@@ -514,6 +514,54 @@ def compute(self, inputs, outputs):
                          "Group (<model>): src_indices has been defined in both "
                          "connect('indep.x', 'C1.x') and add_input('C1.x', ...).")
 
+    def test_incompatible_src_indices_error(self):
+        class ControlInterpComp(om.ExplicitComponent):
+
+            def setup(self):
+                self.add_output('x', shape=(3, 1))
+
+            def compute(self, inputs, outputs):
```
Comment (swryan) [inline openmdao/core/tests/test_group.py:523]: don't really need to redefine these `compute`/`compute_partials` methods, `pass` is the default implementation

File: openmdao/core/tests/test_group.py [hunk 4]
```diff
@@ -514,6 +514,54 @@ def compute(self, inputs, outputs):
...
+
+        with self.assertRaises(ValueError) as context:
+            p.setup()
+
+        self.assertEqual(str(context.exception), msg)
+
+        p.model._raise_connection_errors = False
+
+        with assert_warning(UserWarning, msg):
+            p.setup()
```
Comment (swryan) [inline openmdao/core/tests/test_group.py:563]: did you mean to check the warning message here?

Comment (DKilkenny) [reply openmdao/core/tests/test_group.py:563]: Yes

Comment (swryan) [reply openmdao/core/tests/test_group.py:563]: ah right.. my bad, sorry

**Pull request merged**